src/org/traccar/protocol/NavisProtocolDecoder.java

```diff
@@ -31,14 +35,25 @@
 import java.nio.charset.StandardCharsets;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.Date;
 
 public class NavisProtocolDecoder extends BaseProtocolDecoder {
 
     private String prefix;
     private long deviceUniqueId, serverId;
+    private int flexDataSize; // bytes
+    private int flexBitfieldDataSize; // bits
+    private final byte[] flexBitfield;
+    private static final Logger LOGGER = LoggerFactory.getLogger(NavisProtocolDecoder.class);
```

Comment:
Logger should on top and separated from everything else, especially from instance variables.

Reply:
Fixed